### PR TITLE
Use existing migration token during setup

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

This PR fixes an issue when a migration token exists (set in a constant) but does not get saved to the custom DB scripts. 

### Testing

* [x] This change adds unit test coverage
* [x] This change has been tested on WP 5.1.1

### Checklist

* [x] All existing and new tests complete without errors
* [x] All code quality tools/guidelines in the [Contribution guide](CONTRIBUTION.md) have been run/followed
* [x] All active GitHub CI checks have passed

### CODE REVIEW

File: lib/initial-setup/WP_Auth0_InitialSetup_Consent.php [hunk 1]
```diff
@@ -170,9 +170,12 @@ public function consent_callback( $name ) {
 		if ( $should_create_and_update_connection ) {
 
 			if ( $connection_exists === false ) {
-				$migration_token = JWT::urlsafeB64Encode( openssl_random_pseudo_bytes( 64 ) );
-				$operations      = new WP_Auth0_Api_Operations( $this->a0_options );
-				$response        = $operations->create_wordpress_connection(
+				$migration_token = $this->a0_options->get( 'migration_token' );
+				if ( empty( $migration_token ) ) {
+					$migration_token = JWT::urlsafeB64Encode( openssl_random_pseudo_bytes( 64 ) );
```
Comment (lbalmaceda) [inline lib/initial-setup/WP_Auth0_InitialSetup_Consent.php]: what if this "is empty logic" is part of the `$this->a0_options->get( 'migration_token' );`? would it make sense, can that be done? so that whenever you request it you always obtain a valid value.

File: tests/testInitialSetupConsent.php [hunk 2]
```diff
@@ -204,14 +204,70 @@ public function testThatNewConnectionIsCreatedAndFailedClientGrantRedirects() {
...
+		$this->assertEquals( '/wp-admin/admin.php', $redirect_url['path'] );
+		$this->assertContains( 'page=wpa0-setup', $redirect_url['query'] );
+		$this->assertContains( 'step=2', $redirect_url['query'] );
+		$this->assertContains( 'profile=social', $redirect_url['query'] );
+
+		$this->assertEquals( 'TEST_CLIENT_ID', self::$opts->get( 'client_id' ) );
+		$this->assertEquals( 'TEST_CLIENT_SECRET', self::$opts->get( 'client_secret' ) );
+		$this->assertEquals( 1, self::$opts->get( 'db_connection_enabled' ) );
+		$this->assertEquals( 'TEST_CREATED_CONN_ID', self::$opts->get( 'db_connection_id' ) );
+		$this->assertEquals( 'TEST_MIGRATION_TOKEN', self::$opts->get( 'migration_token' ) );
```
Comment (lbalmaceda) [inline tests/testInitialSetupConsent.php]: Please add a test that asserts that when this token is not set (Empty) a random token (not empty, length >x, etc) is being used. I couldn't find it on this file.

File: lib/initial-setup/WP_Auth0_InitialSetup_Consent.php (hunk 1 above)
Comment (joshcanhelp) [reply lib/initial-setup/WP_Auth0_InitialSetup_Consent.php]: We don't really want that to happen. The value needs to be added to a script JS file and added to the Connection so generating one out of the blue would not be helpful. Also, there are times when we need to know if the setting is empty or not. Otherwise, we'd have to add an `has()` method or similar. These tokens are not generated often enough to make that worth it.

File: tests/testInitialSetupConsent.php (hunk 2 above)
Comment (joshcanhelp) [reply tests/testInitialSetupConsent.php]: See the added line in the test above (which is uses that process)
